src/util/kaldi-table-inl.h

```diff
@@ -298,10 +338,38 @@ template<class Holder>  class SequentialTableReaderScriptImpl:
     }
     std::string line;
     if (getline(script_input_.Stream(), line)) {
-      SplitStringOnFirstSpace(line, &key_, &data_rxfilename_);
-      if (!key_.empty() && !data_rxfilename_.empty()) {
+      std::string data_rxfilename_next, rest;
+      // rest: a temp string which may contain the range specifier.
+      SplitStringOnFirstSpace(line, &key_, &rest);
+      if (!key_.empty() && !rest.empty()) {
         // Got a valid line.
         state_ = kHaveScpLine;
+        // Here we "peel off" the range_ specifer (if any, e.g. [1:2,2:10])
+        // from data_rxfilename_ (e.g. 1.ark:100).
+        if (rest[rest.size()-1] == ']') {
+          std::vector<std::string> splits;
+          SplitStringToVector(rest, "[", true, &splits);
+          if (splits.size() == 2) {
+            data_rxfilename_next = splits[0];
+            range_.assign(splits[1], 0, splits[1].size()-1) ;
+          } else {
+            // Got an invalid line.
+            state_ = kError;  // we can't make sense of this
+            // scp file and will now die.
+          }
+        } else {
+          data_rxfilename_next = rest;
+          range_ = "";
+        }
+        if (old_state == kLoadSucceeded || old_state == kRangeExtracted ||
+            old_state == kRangeExtractionFailed) {
+          if (data_rxfilename_ == data_rxfilename_next) {
+            state_ = kObjectLoaded;
+          } else {
+            holder_.Clear();
+          }
+        }
+        data_rxfilename_ = data_rxfilename_next;
       } else {
         // Got an invalid line.
         state_ = kError;  // we can't make sense of this
```

Comment:
Please make sure that if we go to the state kError, at some point an informative error message is printed.  It's not clear that that was the case in the previous code.